src/client.rs

```diff
@@ -82,6 +82,7 @@ struct Subscription {
     subject: String,
     queue_group: Option<String>,
     messages: channel::Sender<Message>,
+    filter: MessageFilter,
```

Comment:
What is this for? Is this just a filter subject that is attached to a consumer? Struggling to realize what this is here for.

Reply:
Filtering and processing as messages arrive on the write side of the subscription.
Used to handle control messages like heartbeats, flow control, detect sequence mismatches, etc.

Reply:
Renamed to _preprocess_ on both ends (before send in client and internal recv filter in subscription).